docs/docs/specs/v0.2/session-graph.md

```diff
@@ -0,0 +1,292 @@
+
+We agreed to several use-cases that any solution must facilitate.
+
+### Basics
+
+Plugins should be able to add content items and a node under which they can be organized by the plugin that added the content items.
+
+e.g.:
+> `MarkdownPlugin` can add nodes representing documents it finds and associate them with a node that represents itself as the documents’ provenance
+
+#### Adding content items
+
+```tsx
+const MARKDOWN_NS = 'dxos.org/markdown-plugin';
+
+const MARKDOWN_GROUP = `${MARKDOWN_PLUGIN}/documents`;
+
+const handleReady = async () => {
+    
+    // The plugin creates its fallback group node
+  const groupNode: Node = {
+    id: MARKDOWN_GROUP,
+    label: ['plugin name', {ns: MARKDOWN_NS}],
+    params: {
+      provenance: MARKDOWN_NS,
+      icon: 'ArticleMedium'
+    },
+  };
+  
+  // Adds the group node to the graph
+  sessionGraph.addNodes(groupNode);
+  
+  // Gets the entities from its ontological substrate, possibly specifying that it only needs certain parts of the entities
+  const documents = await runQuery(MarkdownSessionGraphQuery);
+  
+  // Creates content nodes from the query results
+  const contentNodes: Node[] = documents.map((document)=>({
+    id: document.id,
+    label: document.title ?? ['untitled document', {ns: MARKDOWN_NS}],
+    params: {
+        provenance: MARKDOWN_NS,
+      icon: document.icon ?? 'ArticleMedium'
+    }
+  }));
+    
+    // Adds the query result nodes to the graph
+  sessionGraph.addNodes(...contentNodes);
+    
+    // Creates the one-to-many child relation from the group node to the content nodes
+  const provenanceChildrenRelation = {
+    of: MARKDOWN_GROUP,
+    by: new Set(documents.map(({id})=>id)),
+    key: 'provenance-child'
+  }
+  
+  // Creates the one-to-one parent relations from the content nodes back to the group node
+  const provenanceParentRelations = documents.reduce((acc, document)=>{
+    acc.push({of: document.id, by: MARKDOWN_GROUP, key: 'provenance-parent'})
+  }, []);
+  
+  // Adds both kinds of relation to the graph
+  sessionGraph.addRelations(provenanceChildrenRelation, ...provenanceParentRelations);
+  
+  // Done!
+}
+```
+
+#### Rendering resolved data & child nodes
+
+```tsx
+const ListItem = ({ id, node }: {id: NodeKey, node: Node}) => {
+  const datum = useResolvedDatum(id);
+  return <Surface role='listitem' datum={datum} />
+}
+
+const List = ({id, node, relationKey = 'provenance-child'}: {id, NodeKey, node: Node, relationKey: string}) => {
+  const datum = useResolvedDatum(id);
+  const childNodes: Map<NodeKey, Node> = useRelatedNodes(id, relationKey);
+  return <>
+    <Surface role='list__meta' datum={datum} />
+    <ul>
+      {childNodes.entries().map((childId, node)=>(
+        <ListItem id={childId} node={node}/>
+      ))}
+    </ul>
+  </>
+}
+```
+
+The hooks `useResolvedDatum` and `useRelatedNodes` leverage signals to propagate updates to the substrate entity or the node’s relations respectively.
```

Comment:
How are you expecting the data to be resolved from a node if it's not directly referenced by the node? Will the plugin need to provide a resolver?

Reply:
Yes, was planning on illustrating that next.